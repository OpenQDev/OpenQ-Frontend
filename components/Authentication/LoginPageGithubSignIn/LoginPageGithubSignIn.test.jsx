import React from 'react';
import LoginPageGithubSignIn from '.';
import renderer from 'react-test-renderer';
import nextRouter from 'next/router';

nextRouter.useRouter = vi.fn();
nextRouter.useRouter.mockImplementation(() => ({
  query: { type: null },

  prefetch: vi.fn(() => {
    return { catch: vi.fn };
  }),
}));

describe('LoginPageGithubSignIn', () => {
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(<LoginPageGithubSignIn />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
