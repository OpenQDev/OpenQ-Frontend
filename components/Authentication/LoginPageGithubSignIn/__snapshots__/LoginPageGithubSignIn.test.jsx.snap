// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginPageGithubSignIn > should match DOM Snapshot 1`] = `
<button
  className="flex justify-center btn-primary  rounded-lg py-2 font-bold whitespace-nowrap w-full"
  onClick={[Function]}
>
  <div
    className="flex flex-row items-center justify-center space-x-3"
  >
    <div
      className="h-4 w-4 md:h-6 md:w-6 relative rounded-full"
    >
      <img
        alt="Picture of the author"
        data-nimg="1"
        decoding="async"
        height={24}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/social-icons/github-logo-white.svg"
        style={
          {
            "color": "transparent",
          }
        }
        width={24}
      />
    </div>
    <div>
      Continue with Github
    </div>
    <svg
      aria-hidden="true"
      className="octicon octicon-arrow-right"
      fill="currentColor"
      focusable="false"
      height={32}
      role="img"
      style={
        {
          "display": "inline-block",
          "overflow": "visible",
          "userSelect": "none",
          "verticalAlign": "text-bottom",
        }
      }
      viewBox="0 0 24 24"
      width={32}
    >
      <path
        d="M13.22 19.03a.75.75 0 0 1 0-1.06L18.19 13H3.75a.75.75 0 0 1 0-1.5h14.44l-4.97-4.97a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l6.25 6.25a.75.75 0 0 1 0 1.06l-6.25 6.25a.75.75 0 0 1-1.06 0Z"
      />
    </svg>
  </div>
</button>
`;
